cle/backends/elf/elf.py

```diff
@@ -754,6 +757,57 @@ def __register_tls(self, seg_readelf):
         self.tls_data_size = seg_readelf.header.p_filesz
         self.tls_data_start = AT.from_lva(seg_readelf.header.p_vaddr, self).to_rva()
 
+    def __register_relro(self, segment_relro):
+        segment_relro = ELFSegment(segment_relro, relro=True)
+
+        def ___segments_overlap(seg1, seg2):
+            # Re-arrange so seg1 starts first
+            seg1, seg2 = (seg1, seg2) if seg1.min_addr < seg2.min_addr else (seg2, seg1)
+            # seg1 and seg2 overlap if seg2 starts before seg1 ends
+            return seg2.min_addr <= seg1.max_addr
+
+        overlapping_segments = [seg for seg in self.segments if ___segments_overlap(segment_relro, seg)]
+
+        if len(overlapping_segments) == 0:
+            l.error("RELRO segment does not overlap with any loaded segment.")
+            return
+
+        if len(overlapping_segments) > 1:
+            # I don't think this ever happens.  If it does, weshould
+            # probably also split the RELRO segment so that each one
+            # has the right permissions in case the two overlapping
+            # segments have different permissions.
+            l.warning("RELRO segment overlaps multiple segments.")
+
+        for overlapping_segment in overlapping_segments:
+            # We will split the overlapping segment into two pieces:
+            # one for the segment below the RELRO segment, and one for
+            # above.
+            segment_below = copy.deepcopy(overlapping_segment)
```

Comment:
is deepcopy safe here? are there backrefs back to the owner or the loader? can you verify that, then add a comment to that effect and add a huge warning comment on the segment object definition?

Reply:
alternately, can we just use the shallow copy?

Reply:
I think you are right that this should just be a shallow copy.  I'll test and verify.